lang/cs/Org.Apache.REEF.Client/YARN/YarnREEFDotNetClient.cs

```diff
@@ -0,0 +1,216 @@
+
+        private SubmitApplication CreateApplicationSubmissionRequest(
+           IJobSubmission jobSubmission,
+           string appId,
+           int maxApplicationSubmissions,
+           JobResource jobResource)
+        {
+            string command = _yarnJobCommandProvider.GetJobSubmissionCommand();
+            Log.Log(Level.Verbose, "Command for YARN: {0}", command);
+            Log.Log(Level.Verbose, "ApplicationID: {0}", appId);
+            Log.Log(Level.Verbose, "MaxApplicationSubmissions: {0}", maxApplicationSubmissions);
+            Log.Log(Level.Verbose, "Driver archive location: {0}", jobResource.RemoteUploadPath);
+
+            var submitApplication = new SubmitApplication
+            {
+                ApplicationId = appId,
+                ApplicationName = jobSubmission.JobIdentifier,
+                AmResource = new Resouce
+                {
+                    MemoryMB = jobSubmission.DriverMemory,
+                    VCores = 1 // keeping parity with existing code
+                },
+                MaxAppAttempts = maxApplicationSubmissions,
+                ApplicationType = REEFApplicationType,
+                KeepContainersAcrossApplicationAttempts = true,
+                Queue = @"default", // keeping parity with existing code
+                Priority = 1, // keeping parity with existing code
+                UnmanagedAM = false,
+                AmContainerSpec = new AmContainerSpec
+                {
+                    LocalResources = new LocalResources
+                    {
+                        Entries = new[]
+                        {
+                            new KeyValuePair<string, LocalResourcesValue>
+                            {
+                                Key = _fileNames.GetReefFolderName(),
+                                Value = new LocalResourcesValue
+                                {
+                                    Resource = jobResource.RemoteUploadPath,
+                                    Type = ResourceType.ARCHIVE,
+                                    Visibility = Visibility.APPLICATION,
+                                    Size = jobResource.ResourceSize,
+                                    Timestamp = jobResource.LastModificationUnixTimestamp
+                                }
+                            }
+                        }
+                    },
+                    Commands = new Commands
+                    {
+                        Command = command
+                    }
+                }
+            };
+
+            return submitApplication;
+        }
+
+        /// <summary>
+        /// Creates the temporary directory to hold the job submission.
+        /// </summary>
+        /// <returns>The path to the folder created.</returns>
+        private string CreateDriverFolder(string jobId, string appId)
+        {
+            return Path.GetFullPath(Path.Combine(Path.GetTempPath(), string.Join("-", "reef", jobId, appId)));
```

Comment:
This is running on the client, which isn't generally managed by a resource manager. Hence, this should be fine.